src/expression/comparison/__test__/unit/eq.test.ts

```diff
@@ -15,24 +15,44 @@ describe('Expression - Comparison - Equal', () => {
     })
   })
 
+  const primitives = [1, '1', true, false, undefined, null]
+  const testCases: [Operand, Operand, boolean][] = [
+    // Truthy cases - type A === type A
+    ...primitives.map<[Operand, Operand, boolean]>((value) => [operand(value), operand(value), true]),
+    // Falsy - different types - across all permutations
+    ...permutation(primitives)
+      .map<[Operand, Operand, boolean]>(([left, right]) => [operand(left), operand(right), false]),
```

Comment:
Neat. I'm surprised you need to `<[Operand, Operand, boolean]>`?

Reply:
because this return from map ```=> [operand(value), operand(value), true]``` = ```[Operand|boolean][] type for TS``` where we are more concrete here, we want to the exact tuple array for test cases

Reply:
You need. Typescript assumes that the return type is `(Operand | boolean)[]`. It doesn't infer that this is a tuple.